modules/core/test/test_arithm.cpp

```diff
@@ -2881,6 +2881,28 @@ TEST(Core_MinMaxIdx, rows_overflow)
     }
 }
 
+TEST(Core_MinMaxIdx, non_continuous)
+{
+    Mat big(32, 32, CV_32FC1, Scalar(15));
+    Mat small = big(Range(8, 8+16), Range(8, 8+16));
+
+    small = 50;
+    small.at<float>(8, 8) = 100;
+    small.at<float>(9, 9) = 25;
```

Comment:
It might be worth using coordinates with x!=y (e.g. 6,7 & 8,9)to ensure the implementation doesn't get them mixed up.